listenbrainz/webserver/static/css/recommendation-page.less

```diff
@@ -0,0 +1,87 @@
+
+
+#recommendations {
+    padding: 10px;
+
+    > .recommendation-card {
+      padding: 15px;
+      margin-bottom: 7px;
+      max-height: 72px;
+
+      &.current-recommendation {
+        background-color: rgba(217, 237, 247, 0.3) !important ;
+      }
+
+       div {
+         padding: 0px;
+       }
+
+      > .col-xs-9, .col-xs-12 {
+
+        > .col-xs-9, .col-xs-12 {
+          display: flex;
+
+          .track-details {
+            width: 100%;
+
+            p {
+              overflow: hidden;
+              text-overflow: ellipsis;
+              white-space:nowrap;
+              margin-bottom: 0px;
+            }
+          }
+        }
+
+      }
+
+      > .col-xs-3 {
+        display: table;
+        height: 40px;
+
+        > * {
+          display: table-cell;
+          vertical-align: middle;
+        }
+
+        a {
+          word-break: break-word;
+        }
+
+        .recommendation-controls {
+
+          > svg:hover {
+            cursor: pointer;
+          }
+
+          .smile, .laugh, .angry, .disappoint, .meh {
+              font-size: 18px;
+              margin: 0% 7%;
```

Comment:
This looks really wide on desktop with percentages
```suggestion
              margin: 0 0.4em;
```

Reply:
![image](https://user-images.githubusercontent.com/26141595/95950885-f9ea1580-0e12-11eb-91a1-bb06b47b99c6.png)

Renders like this if I change the margin.

Reply:
Yeah, that's due to  the `min-width: 400px` style you added to the dropdown-menu.
Without it the space is too small for all emoticons to fit on one line, but we can fix that and have the containing element adjust to the size of the icons, and vice-verse:

```
.recommendation-controls.open .dropdown-menu {
    display: flex;
    // width: 180px; // if you want, you can define a width here, and the elements inside will spread out to fill the space
}
.smile, .laugh, .angry, .disappoint, .meh {
    font-size: 18px;
    flex: 1; // Fills out the space evenly
    text-align: center;
}
```

Reply:
`.recommendation-controls {

        > svg:hover {
          cursor: pointer;
        }

        .recommendation-controls.open .dropdown-menu {
          display: flex;
          width: 1000px;
          
          button {
            width: 100%;
            background: none;
            color: inherit;
            border: none;
            padding: 5px 20px;
            font: inherit;
            cursor: pointer;
            outline: inherit;
            text-align: left;

            &:hover {
              color: #ffffff;
              background-color: #eb743b;
            }
          }
        }

        .smile, .laugh, .angry, .disappoint, .meh {
          font-size: 18px;
          flex: 1;
          text-align: center;
          margin: 0 0.4em;
        }
      }`

Reply:
![image](https://user-images.githubusercontent.com/26141595/95989648-28361800-0e48-11eb-8125-cd706bb1182c.png)

Reply:
Was just experimenting doing `width: 1000px`, it seems the flex isn't expanding.

Reply:
I see two .recommendation-controls, which means your css will look like:
`.recommendation-controls .recommendation-controls.open .dropdown-menu` and that won't affect your element.

Here's my refactor:
```less
.recommendation-controls {

  &.open .dropdown-menu {
    display: flex;
    width: 1000px;
  }

  .smile, .laugh, .angry, .disappoint, .meh {
    font-size: 18px;
    flex: 1;
    text-align: center;
    margin: 0 0.4em;
  }
}
```

The `&.open` means `{parent selector}.open` which in this case will compile as `.recommendation-controls.open`